src/main/java/pl/uj/io/cuteanimals/model/action/TalkAction.java

```diff
@@ -0,0 +1,64 @@
+package pl.uj.io.cuteanimals.model.action;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import pl.uj.io.cuteanimals.model.GameState;
+import pl.uj.io.cuteanimals.model.Result;
+import pl.uj.io.cuteanimals.model.interfaces.IAction;
+import pl.uj.io.cuteanimals.model.interfaces.ICharacter;
+import pl.uj.io.cuteanimals.model.interfaces.ILocation;
+import pl.uj.io.cuteanimals.model.interfaces.IResult;
+
+public class TalkAction implements IAction {
+
+    private List<String> args;
+
+    private ILocation location;
+
+    public TalkAction() {
+        this.args = new ArrayList<>();
+    }
+
+    public TalkAction(List<String> args, ILocation location) {
+        this.args = args;
+        this.location = location;
+    }
+
+    @Override
+    public IResult execute(ICharacter character) {
+        if (args.size() == 0) {
+            return new Result("You want to talk... who?");
+        }
+
+        if (!getAcceptableStates().contains(character.getCurrentGameState())) {
+            return new Result("This action cannot be executed now");
+        }
+
+        var npc =
+                location.getNPCs()
+                        .stream()
+                        .filter(x -> x.getName().equals(args.get(0)))
+                        .collect(Collectors.toList());
+        args.clear();
+
+        return npc.size() >= 1
+                ? new Result(npc.get(0).getQuote())
+                : new Result("This action cannot be executed now");
```

Comment:
Maybe change to sth like:
```suggestion
                : new Result("There is nobody named like that.");
```
Just to keep immersion.

Reply:
You've changed wrong line xD